avocado/core/suite.py

```diff
@@ -0,0 +1,141 @@
+from enum import Enum
+from uuid import uuid1
+
+from .exceptions import OptionValidationError
+from .future.settings import settings
+from .loader import loader, LoaderError, LoaderUnhandledReferenceError
+from .resolver import resolve
+from .tags import filter_test_tags
+from .test import DryRunTest
+from .utils import resolutions_to_tasks
+from .varianter import Varianter
+
+
+class TestSuiteError(Exception):
+    pass
+
+
+class TestSuiteStatus(Enum):
+    RESOLUTION_NOT_STARTED = 0
+    TESTS_NOT_FOUND = 1
+    TESTS_FOUND = 2
+    UNKNOWN = 3
+
+
+class TestSuite:
+    def __init__(self, name, config, tests=None):
+        self.name = name
+        self.tests = tests or None  # this will force None when tests = []
+
+        # Create a complete config dict with all registered options + custom
+        # config
+        self.config = settings.as_dict()
+        if config:
+            self.config.update(config)
+
+        self._variant = None
+        self._references = None
+
+        if config.get('run.dry_run.enabled'):
+            self.convert_to_dry_run()
+
+    def __len__(self):
+        """This is a convenient method to run `len()` over this object.
+
+        With this you can run: len(a_suite) and will return the same as
+        `len(a_suite.tests)`.
+        """
+        return self.size
+
+    def _parse_variant(self):
+        # Varianter not yet parsed, apply configs
+        if not self.variant.is_parsed():
+            try:
+                self.variant.parse(self.config)
+            except (IOError, ValueError) as details:
+                raise OptionValidationError("Unable to parse "
+                                            "variant: %s" % details)
+
+    @property
+    def references(self):
+        if self._references is None:
+            self._references = self.config.get('run.references')
+        return self._references
+
+    @property
+    def size(self):
+        """The overall length/size of this test suite."""
+        if self.tests is None:
+            return 0
+        return len(self.tests)
+
+    @property
+    def status(self):
+        if self.tests is None:
+            return TestSuiteStatus.RESOLUTION_NOT_STARTED
```

Comment:
I like the idea of a status, I only miss some tests on the concept and implementation.

Reply:
I took this idea from this comment: 

https://github.com/avocado-framework/avocado/blob/837883c264eb195e4be742fb5b2d05f548178b20/avocado/core/job.py#L175

So maybe I don't have a clear use case yet, just added there for future use based on this comment. Not sure if we should keep it or not.

Reply:
I think it's valid information (I'd like to have it as a user of a test suite), but you're right that it may not be an urgent need.  I'll leave it to you to keep it (and add some basic tests) or drop it.

Reply:
@clebergnu thanks for this comment. I got some errors after a review. Sending a v2 with a some tests here.